### ISSUE TITLE
cloudinit provider can not handle sensitive variables in content

### ISSUE THREAD

**Issue by Hinni** [hashicorp/terraform-provider-cloudinit#11]
### Community Note
Please vote on this issue by adding a 👍 reaction to the original issue to help the community and maintainers prioritize this request
Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Terraform Version
Terraform v0.14.0
+ provider registry.terraform.io/hashicorp/cloudinit v2.1.0

### Affected Resource(s)
- cloudinit_config

### Terraform Configuration Files
```hcl
variable "service_token" {
  type      = string
  sensitive = true
}

terraform {
  required_version = "~> 0.14"
  required_providers {
    cloudinit = {
      source  = "hashicorp/cloudinit"
      version = "~> 2.1"
    }
  }
}

data "cloudinit_config" "issue" {
    gzip = true
    base64_encode = true

    part {
        content_type = "text/cloud-config"
        content = templatefile("./cloudinit.tmpl", {
            "ServiceToken" = var.service_token
        })
        filename = "init.cfg"
    }
}

File content of cloudinit.tmpl
#cloud-config

runcmd:
 - [ wget, --header="Authorization: Bearer ${ServiceToken}", "https://local.repo/more.zip"]
```

### Debug Output
https://gist.github.com/Hinni/043948cb0fde6fd7d385ace4996d0c60

### Panic Output
no

### Expected Behavior
Successful terraform plan without an error.

### Actual Behavior
Terraform plan fails with:
```hcl
Error: .part[0].content: value has marks, so it cannot be serialized
```

### Steps to Reproduce
1. terraform init
1. terraform plan

### Important Factoids
no

### References
no

**Comment by austinvalle** [comment]
Hi there @Hinni! 👋🏻

Thanks for the bug report and sorry about the delayed response! This bug looked to be related to this Terraform CLI issue here: https://github.com/hashicorp/terraform/issues/27832.

This was fixed in a later version of terraform `v0.14`. Here is a test with your config passing on `v0.14.11`:

#### Recreated bug with `v0.14.0`
![image](https://user-images.githubusercontent.com/8650838/217659769-94f77e8f-0093-4d3e-bb21-865af7493851.png)

#### Fixed with `v0.14.11`
![image](https://user-images.githubusercontent.com/8650838/217659800-bbc5721c-67c9-4201-a578-c52cf17749e3.png)

**Comment by Hinni** [comment]
@austinvalle Thanks for your update on this topic.
